src/app/chat/services/chat/chat.service.ts

```diff
@@ -0,0 +1,147 @@
+import { Injectable } from '@angular/core';
+import { Apollo, ApolloQueryObservable } from 'apollo-angular';
+import { AuthenticationService } from '../../../shared/services/authentication.service';
+import { Subscription } from 'apollo-client';
+import { sendMessageMutation } from '../../../graphql/queries/send-message.mutation';
+import { messagesQuery } from '../../../graphql/queries/messages.query';
+import { chatMessageAddedSubscription } from '../../../graphql/queries/chat-message-added.subscription';
+import { MessagesQuery } from '../../../graphql/types/types';
+import { UserDataService } from '../../../shared/services/user-data/user-data.service';
+
+@Injectable()
+export class ChatService {
+  private messagesObservable: ApolloQueryObservable<MessagesQuery.Result>;
+  private cursor: any;
+  private messagesSubscription: Subscription;
+  private noMoreToLoad = false;
+  private loadingMoreMessages = false;
+  private user = { name: undefined, avatar: undefined };
```

Comment:
it is init in subscribe - optimistic response can be used before subscribe is fired